src/renderable/sprite.js

```diff
@@ -269,30 +268,25 @@
             var ax = w * this.anchorPoint.x, ay = h * this.anchorPoint.y;
             renderer.translate(-ax, -ay);
             
-            if ((this.scaleFlag) || (angle !== 0)) {
-
+            if ((this.scaleFlag) || (this.angle !== 0) || (this._sourceAngle !== 0)) {
                 // translate to the defined anchor point
                 renderer.translate(xpos + ax, ypos + ay);
-                if (angle !== 0) {
-                    renderer.rotate(angle);
+                // rotate
+                if (this.angle !== 0) {
+                    renderer.rotate(this.angle);
                 }
                 // scale
                 if (this.scaleFlag) {
                     renderer.scale(this._scale.x, this._scale.y);
                 }
-
+                renderer.translate(-(xpos+ax), -(ypos+ay));
+                // remove image's TexturePacker/ShoeBox rotation
                 if (this._sourceAngle !== 0) {
-                    // swap w and h for rotated source images
+                    renderer.translate(this.height/2, this.width/2);
+                    renderer.rotate(this._sourceAngle);
+                    renderer.translate(-this.height/2, -this.width/2);
```

Comment:
Two is certainly better than five. Even three is better than five, if you want to avoid a full save/restore between the rotations. Translate is commutative. :smiley: